docs/site/Booting-an-Application.md

```diff
@@ -259,12 +259,117 @@ The options for this are passed in a `services` object on `BootOptions`.
+
+### Configuration Booter
+
+#### Description
+
+Discovers configuration files in `configs` and `dist/configs` folders and binds
+them to the application context using `app.configure()`.
+
+The booter also serves as an extension point to plug in configuration loaders
+that handle different form of files. By default, a configuration loader is added
+to recognize `.config.js`, `.config.json`, `.config.yaml`, and `.config.yml`
+files. Files with the same base name are loaded in the precedence of js, json,
+yaml, and yml extensions. For example, `x.config.js` shadows `x.config.json`.
+
+The js/json/yaml/yaml files export an object containing binding
+key/configuration value pairs.
+
+The `.ts` file will be transpiled to `.js`.
+
+```ts
+export = {
+  'loggers.Log1': {
+    level: 'info',
+    __source: 'js',
+  },
+  'loggers.Log2': {
+    level: 'debug',
+    __source: 'js',
+  },
+};
+```
+
+The configuration can be defined in a json file.
+
+```json
+{
+  "loggers.Log1": {
+    "level": "info",
+    "__source": "json"
+  },
+  "loggers.Log2": {
+    "level": "debug",
+    "__source": "json"
+  }
+}
+```
+
+The `yaml` format is supported too.
+
+```yaml
+loggers.Log1:
+  level: info
+  __source: yaml
+loggers.Log2:
+  level: debug
+  __source: yaml
+```
+
+![config-booter](imgs/config-booter.png)
+
+#### Configuration loaders
+
+Configuration loaders are extensions to the configuration booter.
+
+```ts
+@configurationLoader()
+class MyConfigLoader implements ConfigurationLoader {
+  options = {};
+
+  load(_app: Application, projectRoot: string, files: string[]) {
+    // Directly add configuration bindings to the application
+    app.configure('loggers.Log1').to({
+      level: 'info',
+      __source: 'my-loader',
+    });
+    // Export an object containing configuration values keyed by target binding
+    // keys
+    return {
+      'loggers.Log2': {
+        level: 'debug',
+        __source: 'my-loader',
+      },
+    };
+  }
+}
+
+app.add(createBindingFromClass(MyConfigLoader));
+```
+
+#### Options
+
+The options for this can be passed via `BootOptions` when calling
+`app.boot(options: BootOptions)`.
+
+The options for this are passed in a `configurations` object on `BootOptions`.
+
+Available options on the `configurations` object on `BootOptions` are as
+follows:
+
+| Options      | Type                 | Default                                                        | Description                                                                                                           |
+| ------------ | -------------------- | -------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------- |
+| `dirs`       | `string \| string[]` | `['configs', '../configs]`                                     | Paths relative to projectRoot to look in for Service artifacts                                                        |
```

Comment:
What do you mean when you say `Service artifacts` ? Isn't this capability available to all artifacts that can be loaded and configured?